cmd/opentelemetry-prometheus-sidecar/main.go

```diff
@@ -165,8 +165,9 @@ func Main() bool {
 	promURL, _ := url.Parse(cfg.Prometheus.Endpoint)
 
 	promMon := prometheus.NewMonitor(config.PromReady{
-		Logger:  log.With(logger, "component", "prom_ready"),
-		PromURL: promURL,
+		Logger:                 log.With(logger, "component", "prom_ready"),
+		PromURL:                promURL,
+		ScrapeIntervalDeadline: time.Now().Add(config.DefaultScrapeIntervalWaitPeriod),
```

Comment:
The name `ScrapeIntervalDeadline` left me slightly confused. Can we call this `StartupDelayEffectiveStartTime`? This is the nominal time when Prometheus is starting (it's in the future), from which we'll base the startup delay. We add the interval duration to the effective start time to know when the delay is finished.